filter_plugins/shell_extract.py

```diff
@@ -0,0 +1,52 @@
+import re
```

Comment:
Add comments to the functions in this file.

Reply:
Done.